app/Insightly/Listeners/SyncContact.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Insightly\Listeners;
+
+use App\Domain\Contacts\Contact;
+use App\Domain\Contacts\Events\ContactCreated;
+use App\Domain\Contacts\Events\ContactUpdated;
+use App\Domain\Contacts\Repositories\ContactRepository;
+use App\Insightly\Exceptions\ContactCannotBeUnlinked;
+use App\Insightly\InsightlyClient;
+use App\Insightly\InsightlyMapping;
+use App\Insightly\Repositories\InsightlyMappingRepository;
+use App\Insightly\Resources\ResourceType;
+use App\Insightly\SyncIsAllowed;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Support\Arr;
+use Psr\Log\LoggerInterface;
+
+final class SyncContact implements ShouldQueue
+{
+    use Queueable;
+
+    public function __construct(
+        private readonly InsightlyClient $insightlyClient,
+        private readonly ContactRepository $contactRepository,
+        private readonly InsightlyMappingRepository $insightlyMappingRepository,
+        private readonly LoggerInterface $logger,
+    ) {
+    }
+
+    public function handleContactCreated(ContactCreated $contactCreated): void
+    {
+        $contact = $this->contactRepository->getById($contactCreated->id);
+        if (!SyncIsAllowed::forContact($contact)) {
+            return;
+        }
+
+        $this->storeAndLinkContactAtInsightly($contact);
+
+        $this->logger->info(
+            'Contact created',
+            [
+                'domain' => 'insightly',
+                'contact_id' => $contactCreated->id->toString(),
+            ]
+        );
+    }
+
+    private function storeAndLinkContactAtInsightly(Contact $contact): void
+    {
+        $contactIds = $this->insightlyClient->contacts()->findIdsByEmail($contact->email);
+
+        if (empty($contactIds)) {
+            $contactInsightlyId = $this->insightlyClient->contacts()->create($contact);
+        } else {
+            $contactIds = array_values(Arr::sort($contactIds));
```

Comment:
I vaguely remember that we discussed with @corneelwille to take the contact with the most linked projects and opportunities.

Reply:
Done.